src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -336,6 +337,16 @@ GetFunctionInfo(Oid typeId, Oid accessMethodId, int16 procedureId)
 {
 	FmgrInfo *functionInfo = (FmgrInfo *) palloc0(sizeof(FmgrInfo));
 
+	/* if the type is being dropped, early exit with a null return value */
+	HeapTuple tup = SearchSysCache1(TYPEOID, ObjectIdGetDatum(typeId));
+	if (!HeapTupleIsValid(tup))
+	{
+		ereport(DEBUG4, (errmsg("could not get the catalog entries for type %u",
+								typeId)));
+		return functionInfo;
```

Comment:
If the lookup for the type failed, we do not create function manager info, and hence we fail to sort the shard placements using the comparator functions defined for these types.

This may result in distributed deadlocks as we do not have a way to sort these placements and potentially we will try to drop the shards in an arbitrary order.

Reply:
can we return NULL here and have callers check for it? This 0 initialized struct seems a bit arbitrary.